src/app/portlets/content-types/fields/content-type-fields-add-row/content-type-fields-add-row.component.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { ContentTypeFieldsAddRowComponent } from './content-type-fields-add-row.component';
+import { MessageService } from './../../../../api/services/messages-service';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { TooltipModule } from 'primeng/primeng';
+import { async, ComponentFixture, fakeAsync, tick } from '@angular/core/testing';
+import { DOTTestBed } from '../../../../test/dot-test-bed';
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { Field } from '../';
+import { Observable } from 'rxjs/Observable';
+import { HotkeysService, Hotkey } from 'angular2-hotkeys';
+import { TestHotkeysMock } from './../../../../test/hotkeys-service.mock';
+import { MockMessageService } from './../../../../test/message-service.mock';
+
+describe('ContentTypeFieldsAddRowComponent', () => {
+    let comp: ContentTypeFieldsAddRowComponent;
+    let fixture: ComponentFixture<ContentTypeFieldsAddRowComponent>;
+    let de: DebugElement;
+    let el: HTMLElement;
+    let testHotKeysMock: TestHotkeysMock;
+
+    const messageServiceMock = new MockMessageService({
+        'contenttypes.content.add_rows': 'Add Rows'
+    });
+
+    beforeEach(async(() => {
+        testHotKeysMock = new TestHotkeysMock();
+
+        DOTTestBed.configureTestingModule({
+            declarations: [
+                ContentTypeFieldsAddRowComponent
+            ],
+            imports: [
+                TooltipModule,
+                BrowserAnimationsModule
+            ],
+            providers: [
+                { provide: HotkeysService, useValue: testHotKeysMock },
+                { provide: MessageService, useValue: messageServiceMock }
+            ]
+        });
+
+        fixture = DOTTestBed.createComponent(ContentTypeFieldsAddRowComponent);
+        comp = fixture.componentInstance;
+        de = fixture.debugElement;
+        el = de.nativeElement;
+    }));
+
+    it('should display the add rows button by default', () => {
+        const addRowContainer = de.nativeElement.querySelector('.dot-add-rows__container');
+
+        fixture.detectChanges();
+
+        expect(addRowContainer.classList.contains('dot-add-rows__add')).toEqual(true);
+    });
+
+    it('should display a row after click on AddRows button', () => {
+        fixture.detectChanges();
+        const lis = de.queryAll(By.css('li'));
+
+        const addButton = de.nativeElement.querySelector('.dot-add-rows-button__container button');
+        spyOn(comp, 'selectColumnState');
+        addButton.click();
+
+        expect(lis.length).toEqual(4);
+    });
+
+    it('should escape to previous step', () => {
+        comp.selectColumnState();
+
+        testHotKeysMock.callback(['esc']);
+
+        expect(comp.rowState).toBe('add');
+    });
+
+    it('should be able to use left keyboard to highlight columns', () => {
```

Comment:
In this case you are valitading that the use of the key left never goes less than 0, thats ok to validate but the titleof the test doesn't sad that.

Same with the next test.